src/PrivateDns/PrivateDns/Zones/GetAzurePrivateDnsZone.cs

```diff
@@ -1,51 +1,79 @@
-﻿// ------------------------------------------------------------------------------------------------
-// <copyright file="GetAzurePrivateDnsZone.cs" company="Microsoft Corporation">
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// </copyright>
-// ------------------------------------------------------------------------------------------------
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
 
 namespace Microsoft.Azure.Commands.PrivateDns.Zones
 {
+    using System.Collections.Generic;
     using System.Management.Automation;
     using Microsoft.Azure.Commands.PrivateDns.Models;
     using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 
     /// <summary>
     /// Gets one or more existing zones.
     /// </summary>
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsZone", DefaultParameterSetName = "Default"), OutputType(typeof(PrivateDnsZone))]
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsZone", DefaultParameterSetName = ParameterSetDefault), OutputType(typeof(PSPrivateDnsZone))]
     public class GetAzurePrivateDnsZone : PrivateDnsBaseCmdlet
     {
+        private const string ParameterSetDefault = "Default";
         private const string ParameterSetResourceGroup = "ResourceGroup";
-        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The full name of the private zone (without a terminating dot).")]
+
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The full name of the private zone (without a terminating dot).")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The resource group in which the private zone exists.")]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The resource group in which the private zone exists.")]
```

Comment:
Does Public DNS also have Mandatory=false on resource group parameter for the resourcegroupparamset?
I am not sure if a lot of people search for specific zone by names without specifying resource group.

Reply:
changed it to match public dns